apps/api/src/app/topics/use-cases/add-subscribers/add-subscribers.use-case.ts

```diff
@@ -16,19 +18,28 @@ interface ISubscriberGroups {
 @Injectable()
 export class AddSubscribersUseCase {
   constructor(
+    private createTopic: CreateTopicUseCase,
     private searchByExternalSubscriberIds: SearchByExternalSubscriberIds,
     private topicSubscribersRepository: TopicSubscribersRepository,
     private topicRepository: TopicRepository
   ) {}
 
   async execute(command: AddSubscribersCommand): Promise<Omit<ISubscriberGroups, 'subscribersAvailableToAdd'>> {
-    const topic = await this.topicRepository.findTopicByKey(
+    let topic = await this.topicRepository.findTopicByKey(
       command.topicKey,
       command.organizationId,
       command.environmentId
     );
+
     if (!topic) {
-      throw new NotFoundException(`Topic with key ${command.topicKey} not found in current environment`);
+      const createTopicCommand = CreateTopicCommand.create({
+        environmentId: command.environmentId,
+        key: command.topicKey,
+        // TODO: Maybe make more clear that is a provisional name
+        name: `Topic-On-The-Fly-${command.topicKey}`,
```

Comment:
My suggestion is either use `topicKey` or keeping this at null value. With a preference to null, since if the user display this name somewhere better to be empty than a strange name in my opinion. Wdyt?

Reply:
I wanted to allow a way to help users to differentiate quickly how a topic was created in case they created it by error not being aware of the on the fly functionality.

Reply:
Agreement to use the substring `Autogenerated` to notify users of the origin of the topic creation.